main.tf

```diff
@@ -62,7 +63,7 @@ resource "aws_apigatewayv2_stage" "default" {
   count = var.create && var.create_default_stage ? 1 : 0
 
   api_id      = aws_apigatewayv2_api.this[0].id
-  name        = "$default"
+  name        = var.default_stage_name
```

Comment:
I thought `$default` is the only valid name for the `default stage` (controlled via `var.create_default_stage`) according to AWS, no?

Reply:
let me double check, I thought I saw another option

Reply:
I think the confusion is between `stage` and `route` - the default route is `$default` but stage can be anything